src/jit/codegenarm64.cpp

```diff
@@ -3409,11 +3411,38 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
 
         if (tree->gtFlags & GTF_IND_VOLATILE)
         {
-            // issue a full memory barrier a before volatile StInd
-            instGen_MemoryBarrier();
+            bool useStoreRelease = (dataReg >= REG_INT_FIRST) && (dataReg <= REG_ZR) && !addr->isContained() &&
+                                   ((attr == EA_1BYTE) || !(tree->gtFlags & GTF_IND_UNALIGNED));
```

Comment:
Isn't there already a method that validates  that we have a General purpose register that we can use for this ?
`(dataReg >= REG_INT_FIRST) && (dataReg <= REG_ZR)`

We don't want to bake in the fact that REG_ZR has the largest enum value for a general purpose register in this code.